### PR TITLE
zita-resampler: fix build for aarch64

### PR DESCRIPTION
**Pull request by risicle**
###### Motivation for this change
ZHF #144627

Also remove `-march=native` for reproducibility

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/audio/zita-resampler/default.nix [hunk 1]
```diff
@@ -15,7 +15,10 @@ stdenv.mkDerivation rec {
 
   patchPhase = ''
     cd source
-    sed -e "s@ldconfig@@" -i Makefile
+    substituteInPlace Makefile \
+      --replace 'ldconfig' "" \
+      --replace '-march=native' "" \
+      --replace '-DENABLE_SSE2' ""
```
Comment (thiagokokada) [inline pkgs/development/libraries/audio/zita-resampler/default.nix:21]: I think it is better to keep this flag on `x86_64` systems.

Comment (risicle) [reply pkgs/development/libraries/audio/zita-resampler/default.nix:21]: Certainly sse2 should be present for all x86_64 machines, but I was thinking that the compiler would set this automatically. Thinking again, I'm not sure this is a standard flag so should probably turn it back on for x86_64 :+1:

File: pkgs/development/libraries/audio/zita-resampler/default.nix [hunk 2]
```diff
@@ -15,7 +15,10 @@ stdenv.mkDerivation rec {
 
   patchPhase = ''
     cd source
-    sed -e "s@ldconfig@@" -i Makefile
+    substituteInPlace Makefile \
+      --replace 'ldconfig' "" \
+      --replace '-march=native' "" \
```
[the remark this answers is not in the document]
Comment (risicle) [reply pkgs/development/libraries/audio/zita-resampler/default.nix:20]: Oh good.

File: pkgs/development/libraries/audio/zita-resampler/default.nix (hunk 1 above)
Comment (thiagokokada) [reply pkgs/development/libraries/audio/zita-resampler/default.nix:22]: Yeah, this looks like a macro flag that should be used by the code internally.

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 145259` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>drumgizmo</li>
    <li>ebumeter</li>
    <li>guitarix</li>
    <li>infamousPlugins</li>
    <li>kapitonov-plugins-pack</li>
    <li>xtuner</li>
    <li>zita-ajbridge</li>
    <li>zita-at1</li>
    <li>zita-njbridge</li>
    <li>zita-resampler</li>
  </ul>
</details>

**Review by thiagokokada — DISMISSED** [review]

File: pkgs/development/libraries/audio/zita-resampler/default.nix [hunk 3]
```diff
@@ -15,7 +15,11 @@ stdenv.mkDerivation rec {
 
   patchPhase = ''
     cd source
-    sed -e "s@ldconfig@@" -i Makefile
+    substituteInPlace Makefile \
+      --replace 'ldconfig' ""
+  '' + lib.optionalString stdenv.targetPlatform.isx86_64 ''
```
Comment (thiagokokada) [inline pkgs/development/libraries/audio/zita-resampler/default.nix:20]: Shouldn't it be?
```suggestion
  '' + lib.optionalString !stdenv.targetPlatform.isx86_64 ''
```

Comment (risicle) [reply pkgs/development/libraries/audio/zita-resampler/default.nix:20]: That would make a lot more sense.

File: pkgs/development/libraries/audio/zita-resampler/default.nix [hunk 4]
```diff
@@ -15,7 +15,11 @@ stdenv.mkDerivation rec {
 
   patchPhase = ''
     cd source
-    sed -e "s@ldconfig@@" -i Makefile
+    substituteInPlace Makefile \
+      --replace 'ldconfig' ""
+  '' + lib.optionalString !stdenv.targetPlatform.isx86_64 ''
```
Comment (thiagokokada) [inline pkgs/development/libraries/audio/zita-resampler/default.nix:20]:
```suggestion
  '' + lib.optionalString (!stdenv.targetPlatform.isx86_64) ''
```

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 145259` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>drumgizmo</li>
    <li>ebumeter</li>
    <li>guitarix</li>
    <li>infamousPlugins</li>
    <li>kapitonov-plugins-pack</li>
    <li>xtuner</li>
    <li>zita-ajbridge</li>
    <li>zita-at1</li>
    <li>zita-njbridge</li>
    <li>zita-resampler</li>
  </ul>
</details>

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 145259` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>drumgizmo</li>
    <li>guitarix</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>ebumeter</li>
    <li>infamousPlugins</li>
    <li>kapitonov-plugins-pack</li>
    <li>xtuner</li>
    <li>zita-ajbridge</li>
    <li>zita-at1</li>
    <li>zita-njbridge</li>
    <li>zita-resampler</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**